packages/odyssey-react-mui/src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { forwardRef, useState, useMemo } from "react";
+import type { Ref, ChangeEvent, MouseEvent } from "react";
+import type { OutlinedInputProps, TooltipProps } from "@mui/material";
+import {
+  Tooltip,
+  IconButton,
+  Box,
+  InputLabel,
+  OutlinedInput,
+  InputAdornment,
+} from "@mui/material";
+import { Visibility, VisibilityOff } from "@mui/icons-material";
+import { useOid } from "../../utils";
+
+interface State {
+  password: string;
+  showPassword: boolean;
+}
+
+export interface PasswordInputProps
+  extends Omit<OutlinedInputProps, "type" | "label" | "defaultValue" | "ref"> {
+  ref?: Ref<HTMLInputElement>;
+  defaultValue?: string;
+  label: string;
+  tooltipLabel?:
+    | TooltipProps["title"]
+    | ((isHidden: boolean) => TooltipProps["title"]);
+}
+
+export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
+  (props, ref) => {
+    const {
+      tooltipLabel,
+      id,
+      label,
+      defaultValue: password = "",
+      inputProps,
+      ...rest
+    } = props;
+
+    const [values, setValues] = useState<State>({
+      password,
+      showPassword: false,
+    });
+
+    const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+      setValues({ ...values, password: event.target.value });
+      props.onChange?.(event);
+    };
+
+    const handleClickShowPassword = () => {
+      setValues({
+        ...values,
+        showPassword: !values.showPassword,
+      });
+    };
+
+    const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+    };
+
+    const tooltipTitle = useMemo(() => {
+      return typeof tooltipLabel === "function"
+        ? tooltipLabel(values.showPassword === false)
+        : tooltipLabel;
```

Comment:
Why would the tooltip label ever be a function? Is that because it's a component? If so, components are also classes. Will this work with class components? React has something like a `React.isComponent` function. I don't remember exactly which.

Reply:
Ah, I see. That would be something like `getTooltipLabel` right? Then we can explicitly type it and call it if it exists, otherwise fallback to the text label.

But instead of managing this ourselves, why not only accept a string and let consumers write this logic themselves? If we provide an `onChange`, they can use any standard form or UI logic library to manage this label changing.

Or just accept 2 separate props. Seems like over responsibility for us to manage the label text in `PasswordInput` when it's up to users to figure that out.